examples/widgets/chart/lv_chart_example_8.c

```diff
@@ -0,0 +1,115 @@
+    if (dsc->part == LV_PART_ITEMS)
+    {
+        if (!dsc->p1 || !dsc->p2)
+            return;
+
+        /*Add a line mask that keeps the area below the line*/
+        lv_draw_mask_line_param_t line_mask_param;
+        lv_draw_mask_line_points_init(&line_mask_param, dsc->p1->x, dsc->p1->y, dsc->p2->x, dsc->p2->y, LV_DRAW_MASK_LINE_SIDE_BOTTOM);
+        int16_t line_mask_id = lv_draw_mask_add(&line_mask_param, NULL);
+
+        /*Draw a rectangle that will be affected by the mask*/
+        lv_draw_rect_dsc_t draw_rect_dsc;
+        lv_draw_rect_dsc_init(&draw_rect_dsc);
+        draw_rect_dsc.bg_opa = LV_OPA_COVER;
+        draw_rect_dsc.bg_color = dsc->line_dsc->color;
+
+        lv_area_t a;
+        a.x1 = dsc->p1->x;
+        a.x2 = dsc->p2->x - 1;
+        a.y1 = LV_MIN(dsc->p1->y, dsc->p2->y);
+        a.y2 = obj->coords.y2 - 13; /* -13 cuts off where the rectangle draws over the chart margin. Without this an area of 0 doesnt look like 0 */
+        lv_draw_rect(&a, dsc->clip_area, &draw_rect_dsc);
+
+        /*Remove the mask*/
+        lv_draw_mask_free_param(&line_mask_param);
+        lv_draw_mask_remove_id(line_mask_id);
+    }
+}
+
+/**
+ * Stacked area chart
+ */
+void lv_example_chart_8(void)
+{
+    /*Create a stacked_area_chart.obj*/
+    stacked_area_chart.obj = lv_chart_create(lv_scr_act());
+    lv_obj_set_size(stacked_area_chart.obj, 200, 150);
+    lv_obj_center(stacked_area_chart.obj);
+    lv_chart_set_type(stacked_area_chart.obj, LV_CHART_TYPE_LINE);
+    lv_chart_set_div_line_count(stacked_area_chart.obj, 5, 7);
+    lv_obj_add_event_cb(stacked_area_chart.obj, draw_event_cb, LV_EVENT_DRAW_PART_BEGIN, NULL);
+
+    /* Set range to 0 to 100 for percentages. Draw ticks */
+    lv_chart_set_range(stacked_area_chart.obj,LV_CHART_AXIS_PRIMARY_Y,0,100);
+    lv_chart_set_axis_tick(stacked_area_chart.obj, LV_CHART_AXIS_PRIMARY_Y, 3, 0, 5, 1, true, 30);
+
+    /*Set point size to 0 so the lines are smooth */
+    lv_obj_set_style_size(stacked_area_chart.obj, 0, LV_PART_INDICATOR);
+
+    /*Add some data series*/
+    stacked_area_chart.series_list[0] = lv_chart_add_series(stacked_area_chart.obj, lv_palette_main(LV_PALETTE_RED), LV_CHART_AXIS_PRIMARY_Y);
+    stacked_area_chart.series_list[1] = lv_chart_add_series(stacked_area_chart.obj, lv_palette_main(LV_PALETTE_BLUE), LV_CHART_AXIS_PRIMARY_Y);
+    stacked_area_chart.series_list[2] = lv_chart_add_series(stacked_area_chart.obj, lv_palette_main(LV_PALETTE_GREEN), LV_CHART_AXIS_PRIMARY_Y);
+
+    for (int i = 0; i < 10; i++)
+    {
+        /* Make some random data */
+        int32_t vals[3] = {lv_rand(10, 20), lv_rand(20, 30), lv_rand(20, 30) - i};
+
+        int32_t total = vals[0] + vals[1] + vals[2];
+        float heights[3];
+        int32_t draw_heights[3];
+        int32_t int_sum = 0;
+        float float_sum = 0;
+
+        /* Cascade rounding ensures percentages add to 100 */
+        for (int32_t i = 0; i < 3; i++)
```

Comment:
Could you please pick some more descriptive names for these `i` variables? The outer one could be renamed to `point` and the inner two can be `series_i`. It makes the code more self-documenting and also fixes the shadowing error.